docs/testing/overview.md

```diff
@@ -1,6 +1,52 @@
 # Testing Overview
 
-At MetaMask, we are committed to delivering high-quality software. A key aspect of achieving this goal is through a robust testing strategy that includes both end-to-end (e2e), integration and unit tests. This document provides comprehensive guidelines on each testing type, emphasizing the importance of choosing the right test for the right scenario and detailing the specific practices for writing and organizing these tests.
+At MetaMask, we are committed to delivering high-quality software. These guidelines describe how we use automated tests to maximize product quality while minimizing the cost of writing and maintaining tests.
+
+Our automated tests fall into three general categories: end-to-end (e2e), integration and unit tests. This document provides comprehensive guidelines on each testing type, emphasizing the importance of choosing the right test for the right scenario and detailing the specific practices for writing and organizing these tests.
+
+## Testing principles
+
+What makes a good test? Here are some general principles.
+
+### 1. Tests should describe expectations
+
+What's the difference between a bug and a feature? A feature is expected, a bug is not.
+
+If it's not clear what the code-under-test is supposed to do, it won't be clear what to do when the test fails either. Clear expectations are essential for an effective test.
```

Comment:
A test should ideally focus on one specific behavior or feature, reducing ambiguity in what failed if it does.

Reply:
Generally I think this is great advice, but I don't think it's precisely true to say that we want this for _all_ types of automated tests. End-to-end tests for example would typically have a broader scope, not targeted to a specific feature.

Reply:
Agree with @Gudahtt here... Tests should reveal intent (and as so clearly describe the expectations). Then depending on the type of test, the intent can be beyond a specific feature or behaviour. Usually for unit tests, I would agree with the statement, but for e2e tests for example I would rather focus the tests on use cases (and those can expand through multiple features/behaviours).